### PR TITLE
Fixed react warnings, added check to error on React warnings

### LINKED ISSUE #909: Fail the build when there are React warnings in test log output

### LINKED ISSUE #910: React warnings
From my npm test output:

      App
        ✓ clears profile, ui, and dashboard after unmounting (214ms)
        profile completeness
    Warning: Unsupported style property background-color. Did you mean backgroundColor? Check the render method of `EnhancedSwitch`.
          ✓ redirects to /profile if profile is not complete (284ms)
          ✓ redirects to /profile if profile is not filled out (258ms)
    Warning: Each child in an array or iterator should have a unique "key" prop. Check the render method of `ProfilePage`. See https://fb.me/react-warning-keys for more information.
          ✓ redirects to /profile and goes to the employment step if a field is missing there (139ms)
          ✓ redirects to /profile and goes to the privacy step if a field is missing there (158ms)
          ✓ redirects to /profile and goes to the education step if a field is missing there (240ms)

And below:

        makeProfileProgressDisplay
    Warning: Each child in an array or iterator should have a unique "key" prop. Check the top-level render call using <svg>. See https://fb.me/react-warning-keys for more information.
          ✓ renders the right active display

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?
Fixes #910 
Fixes #909

#### What's this PR do?
Fixes some React warnings and adds a check to fail the build if they come up again

#### Where should the reviewer start?
js_test.sh

#### How should this be manually tested?
Try undoing one of the fixes and running the tests. All tests should pass but the command should return a non-zero value due to the warning, failing the tests overall.

### CODE REVIEW

**Comment by aliceriot** [comment]
👍 

looks like this works! I removed the change to the `backgroundColor` attribute, and the tests overall had a failing exit code, even though all of the tests themselves passed.

File: js_test.sh [hunk 1]
```diff
@@ -1,5 +1,10 @@
 #!/bin/bash
 set -euf -o pipefail
 
-node ./node_modules/mocha/bin/_mocha --compilers js:babel-register static/js/global_init.js ${1:-'static/**/*/*_test.js'}
+node ./node_modules/mocha/bin/_mocha --compilers js:babel-register static/js/global_init.js ${1:-'static/**/*/*_test.js'} 2> >(tee npm-test.err)
```
Comment (aliceriot) [inline js_test.sh]: should we delete the npm-test.err file afterwards though?

Comment (noisecapella) [inline js_test.sh]: Probably a good idea

Comment (aliceriot) [inline js_test.sh]: although it looks like a successful error-free test run will zero the file, so that the `-s` test will pass anyway?

**Comment by aliceriot** [comment]
anyway, still a :+1: from me :D

**Pull request merged**